src/Microsoft.AspNet.OData.Shared/Query/SkipTokenQueryOption.cs

```diff
@@ -0,0 +1,352 @@
+        {
+            if (context == null)
+            {
+                throw Error.ArgumentNull("context");
+            }
+
+            if (String.IsNullOrEmpty(rawValue))
+            {
+                throw Error.ArgumentNullOrEmpty("rawValue");
+            }
+
+            if (queryOptionParser == null)
+            {
+                throw Error.ArgumentNull("queryOptionParser");
+            }
+
+            Context = context;
+            RawValue = rawValue;
+            ParseKeyValuePairs(rawValue);
+            Validator = SkipTokenQueryValidator.GetSkipTokenQueryValidator(context);
+            _queryOptionParser = queryOptionParser;
+        }
+
+        // This constructor is intended for unit testing only.
+        internal SkipTokenQueryOption(string rawValue, ODataQueryContext context)
+        {
+            if (context == null)
+            {
+                throw Error.ArgumentNull("context");
+            }
+
+            if (String.IsNullOrEmpty(rawValue))
+            {
+                throw Error.ArgumentNullOrEmpty("rawValue");
+            }
+
+            Context = context;
+            RawValue = rawValue;
+            Validator = SkipTokenQueryValidator.GetSkipTokenQueryValidator(context);
+            _queryOptionParser = new ODataQueryOptionParser(
+                context.Model,
+                context.ElementType,
+                context.NavigationSource,
+                new Dictionary<string, string> { { "$skiptoken", rawValue } },
+                context.RequestContainer);
+        }
+
+        private void ParseKeyValuePairs(string rawValue)
+        {
+            _propertyValuePairs = new Dictionary<string, object>();
+            string[] keyValues = rawValue.Split(',');
+            foreach (string keyAndValue in keyValues)
+            {
+                string[] pieces = keyAndValue.Split(new char[] { ':' }, 2);
+                if (pieces.Length > 1 && !String.IsNullOrWhiteSpace(pieces[1]))
+                {
+                    IEdmTypeReference type = EdmLibHelpers.GetTypeReferenceOfProperty(Context.Model, Context.ElementClrType, pieces[0]);
+                    object value = ODataUriUtils.ConvertFromUriLiteral(pieces[1], ODataVersion.V401, Context.Model, type);
+                    if (!String.IsNullOrWhiteSpace(pieces[0]))
+                    {
+                        _propertyValuePairs.Add(pieces[0], value);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the given <see cref="ODataQueryContext"/>.
+        /// </summary>
+        public ODataQueryContext Context { get; private set; }
+
+        /// <summary>
+        /// Gets the raw $skiptoken value.
+        /// </summary>
+        public string RawValue { get; private set; }
+
+        /// <summary>
+        /// Gets the value of the $skiptoken as a parsed integer.
```

Comment:
>integer [](start = 57, length = 7)

why integer?

Reply:
Incorrect documentation, leftover from the prototype implementation.